velox/functions/prestosql/Probability.h

```diff
@@ -111,13 +113,44 @@ struct CauchyCDFFunction {
       result = 1.0;
     } else {
       VELOX_USER_CHECK_GE(scale, 0, "scale must be greater than 0");
-
       boost::math::cauchy_distribution<> dist(median, scale);
       result = boost::math::cdf(dist, value);
     }
   }
 };
 
+template <typename T>
+struct InverseCauchyCDFFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void
+  call(double& result, double median, double scale, double probability) {
+    static constexpr double kInf = std::numeric_limits<double>::infinity();
+    static constexpr double kDoubleMax = std::numeric_limits<double>::max();
+
+    if (std::isnan(median) || std::isnan(scale) || std::isnan(probability)) {
+      result = std::numeric_limits<double>::quiet_NaN();
+    } else if (median == kInf || median == kDoubleMax) {
+      result = -kInf;
+    } else if (scale == kInf) {
+      result = median;
+    } else if (probability == 1.0) {
+      result = kInf;
+    } else if (scale <= 0) {
+      result = std::numeric_limits<double>::quiet_NaN();
+      VELOX_USER_CHECK_GT(scale, 0, "scale must be greater than 0");
```

Comment:
If the scale <= 0 as checked by the if condition, then the CHECK is always going to be hit. This code is very confusing. The CHECK should be for the code in the else part of the condition if at all.

Please can you double check.